linked-dependencies/matrix-react-sdk/src/SecurityManager.ts

```diff
@@ -417,3 +423,52 @@ async function doAccessSecretStorage(func: () => Promise<void>, forceReset: bool
         throw e;
     }
 }
+
+// FIXME: this function name is a bit of a mouthful
+export async function tryToUnlockSecretStorageWithDehydrationKey(client: MatrixClient): Promise<void> {
```

Comment:
j'ai du mal à voir le lien entre le retablissement de cette fonction et le but de la PR, n'est-ce pas un résidu d'un autre commit?

Reply:
Cette méthode est utilisé dans le fallback libolm, qui a donc été supprimé lors du clean pour laissé que la crypto rust. Cf `MatrixClientPeg.ts`  :
```

        // fall back to the libolm layer.
        try {
            // check that we have a version of the js-sdk which includes initCrypto
            if (this.matrixClient.initCrypto) {
                await this.matrixClient.initCrypto();
                this.matrixClient.setCryptoTrustCrossSignedDevices(
                    !SettingsStore.getValue("e2ee.manuallyVerifyAllSessions"),
                );
                await tryToUnlockSecretStorageWithDehydrationKey(this.matrixClient);
                StorageManager.setCryptoInitialised(true);
            }
        } catch (e) {
            if (e instanceof Error && e.name === "InvalidCryptoStoreError") {
                // The js-sdk found a crypto DB too new for it to use
                Modal.createDialog(CryptoStoreTooNewDialog);
            }
            // this can happen for a number of reasons, the most likely being
            // that the olm library was missing. It's not fatal.
            logger.warn("Unable to initialise e2e", e);
        }
```